product_pricelist_revision/wizards/pricelist_duplicate_wizard.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2018 Carlos Dauden - Tecnativa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import fields, models
+
+
+class ProductPricelistItemDuplicateWizard(models.TransientModel):
+    _name = "product.pricelist.item.duplicate.wizard"
+    _description = "Wizard Product Pricelist Item Duplicate"
+
+    date_start = fields.Datetime(required=True)
+    date_end = fields.Datetime()
+    variation_percent = fields.Float(digits="Product Price", string="Variation %")
+
+    def action_apply(self):
+        PricelistItem = self.env["product.pricelist.item"]
+        new_items = PricelistItem
+        for item in PricelistItem.browse(self.env.context["active_ids"]):
+            new_items |= item.copy(
+                {
+                    "date_start": self.date_start,
+                    "date_end": self.date_end,
+                    "previous_item_id": item.id,
+                    "fixed_price": item.fixed_price
+                    * (1.0 + self.variation_percent / 100.0),
+                }
+            )
+            item.date_end = self.date_start - relativedelta(seconds=1)
+
+        xmlid = "product_pricelist_revision.product_pricelist_item_action"
+        action = self.env["ir.actions.act_window"]._for_xml_id(xmlid)
```

Comment:
Once the new rules are created, they show the new records, but the action has a context that activates the `Current` filter. If the pricelist is in the future, it can confuse the user since nothing will be displayed. Please remove this key from the context to avoid confusion.

![image](https://github.com/user-attachments/assets/a317a38b-17cf-478f-a0c5-00e3e2febb6d)

![image](https://github.com/user-attachments/assets/5215fb91-197c-4642-ab32-8ce33ecfcdf3)

Reply:
Just here, try something like:

```suggestion
        action = self.env["ir.actions.act_window"]._for_xml_id(xmlid)
        action["context"] = {}
```